increased typesafety and null checks added

imvikashdev opened the pull request.

Neon-20 commented:
Why have you added so many .gitignores, can you explain your PR properly.
@imvikashdev

imvikashdev commented:
I have added js environment gitignore so it works with all project structure, i do it for all my projects, and now it ignores .next folder which is not needed for remote it contains only build

Other than that I have fixed types and added null check if there are valid credentials then only example will start and also added checks in other places
